src/System.Security.Cryptography.Encoding/src/Internal/Cryptography/OpenSslAsnFormatter.cs

```diff
@@ -51,12 +51,17 @@ protected override string FormatNative(Oid oid, byte[] rawData, bool multiLine)
                     int printLen = Interop.Crypto.GetMemoryBioSize(bio);
 
                     // Account for the null terminator that it'll want to write.
-                    StringBuilder builder = new StringBuilder(printLen + 1);
-                    Interop.Crypto.BioGets(bio, builder, builder.Capacity);
+                    var buf = new byte[printLen + 1];
+                    int read = Interop.Crypto.BioGets(bio, buf, buf.Length);
 
-                    return builder.ToString();
+                    if (read < 0)
+                    {
+                        throw Interop.Crypto.CreateOpenSslCryptographicException();
+                    }
```

Comment:
`BIO_s_mem()`-based BIOs return the position of the null terminator, so the empty string returns 0.  File-based BIOs return the position of the null terminator, or 0 on error.  So 0 is ambiguous.   Negative returns are possible, for malfunctioning BIOs.